packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart

```diff
@@ -168,24 +168,25 @@ class CpuProfileData {
     _cpuProfileRoot = CpuStackFrame.root(profileMetaData);
   }
 
-  factory CpuProfileData.parse(Map<String, dynamic> json) {
+  factory CpuProfileData.parse(Map<String, dynamic> json_) {
+    final json = _CpuProfileDataJson(json_);
     final profileMetaData = CpuProfileMetaData(
-      sampleCount: json[sampleCountKey] ?? 0,
-      samplePeriod: json[samplePeriodKey] ?? 0,
-      stackDepth: json[stackDepthKey] ?? 0,
-      time: (json[timeOriginKey] != null && json[timeExtentKey] != null)
+      sampleCount: json.sampleCount ?? 0,
+      samplePeriod: json.samplePeriod ?? 0,
+      stackDepth: json.stackDepth ?? 0,
+      time: (json.timeOriginMicros != null && json.timeExtentMicros != null)
           ? (TimeRange()
-            ..start = Duration(microseconds: json[timeOriginKey])
+            ..start = Duration(microseconds: json.timeOriginMicros!)
             ..end = Duration(
-              microseconds: json[timeOriginKey] + json[timeExtentKey],
+              microseconds: json.timeOriginMicros! + json.timeExtentMicros!,
             ))
           : null,
     );
 
     // Initialize all stack frames.
     final stackFrames = <String, CpuStackFrame>{};
     final Map<String, Object?> stackFramesJson =
-        jsonDecode(jsonEncode(json[stackFramesKey] ?? <String, Object?>{}));
+        jsonDecode(jsonEncode(json.stackFrames ?? <String, Object?>{}));
```

Comment:
@kenzieschmoll do you remember why we are jsonDecoding + encoding? those are expensive operations that are definitely not worth it if we are just trying to get a nice default value for the case where there are no frames. Aside from this decode+encode I don't see anything that is clearly expensive enough that we can't do the parsing more lazily. More lazy parsing might also be faster as right now we parse all the fields and I don't know if we need all of them right away or just a few.